src/utils/formHelperFunctions.ts

```diff
@@ -437,3 +437,35 @@ export function getInitialStep({
     (getOrigin as any)(steps).key
   );
 }
+
+export function castVal(servarType: string | undefined, val: any) {
+  // If there is no type it is a hidden field and we will treat it as a string
+  if (servarType === undefined) return String(val);
+  let castVal = val;
+  switch (servarType) {
+    case 'currency':
+    case 'integer_field':
+    case 'rating':
+    case 'slider':
+      castVal = Number(val);
+      break;
+    case 'checkbox':
+      castVal = !['False', 'false'].includes(val);
+      break;
+    default:
+      castVal = String(val);
+      break;
+  }
+
+  return castVal;
+}
+
+export function getServarTypeMap(steps: any) {
```

Comment:
not needed to address in this PR, but this file has grown massive... probably worth spreading out at some point